mailpoet/lib/Automation/Integrations/WooCommerce/Fields/OrderFieldsFactory.php

```diff
@@ -0,0 +1,347 @@
+          }
+        ),
+        new Field(
+          'woocommerce:order:shipping-phone',
+          Field::TYPE_STRING,
+          __('Shipping phone', 'mailpoet'),
+          function (OrderPayload $payload) {
+            return $payload->getOrder()->get_shipping_phone();
+          }
+        ),
+        new Field(
+          'woocommerce:order:shipping-city',
+          Field::TYPE_STRING,
+          __('Shipping city', 'mailpoet'),
+          function (OrderPayload $payload) {
+            return $payload->getOrder()->get_shipping_city();
+          }
+        ),
+        new Field(
+          'woocommerce:order:shipping-postcode',
+          Field::TYPE_STRING,
+          __('Shipping postcode', 'mailpoet'),
+          function (OrderPayload $payload) {
+            return $payload->getOrder()->get_shipping_postcode();
+          }
+        ),
+        new Field(
+          'woocommerce:order:shipping-state',
+          Field::TYPE_STRING,
+          __('Shipping state/county', 'mailpoet'),
+          function (OrderPayload $payload) {
+            return $payload->getOrder()->get_shipping_state();
+          }
+        ),
+        new Field(
+          'woocommerce:order:shipping-country',
+          Field::TYPE_STRING,
+          __('Shipping country', 'mailpoet'),
+          function (OrderPayload $payload) {
+            return $payload->getOrder()->get_shipping_country();
+          }
+        ),
+        new Field(
+          'woocommerce:order:created-date',
+          Field::TYPE_DATETIME,
+          __('Created date', 'mailpoet'),
+          function (OrderPayload $payload) {
+            return $payload->getOrder()->get_date_created();
+          }
+        ),
+        new Field(
+          'woocommerce:order:paid-date',
+          Field::TYPE_DATETIME,
+          __('Paid date', 'mailpoet'),
+          function (OrderPayload $payload) {
+            return $payload->getOrder()->get_date_paid();
+          }
+        ),
+        new Field(
+          'woocommerce:order:customer-note',
+          Field::TYPE_STRING,
+          __('Customer provided note', 'mailpoet'),
+          function (OrderPayload $payload) {
+            return $payload->getOrder()->get_customer_note();
+          }
+        ),
+        new Field(
+          'woocommerce:order:payment-method',
+          Field::TYPE_ENUM,
+          __('Payment method', 'mailpoet'),
+          function (OrderPayload $payload) {
+            return $payload->getOrder()->get_payment_method();
+          },
+          [
+            'options' => $this->getOrderPaymentOptions(),
+          ]
+        ),
+        new Field(
+          'woocommerce:order:status',
+          Field::TYPE_ENUM,
+          __('Status', 'mailpoet'),
+          function (OrderPayload $payload) {
+            return $payload->getOrder()->get_status();
+          },
+          [
+            'options' => $this->getOrderStatusOptions(),
+          ]
+        ),
```

Comment:
I am a bit afraid of the `wc-` prefix and if `get_status()` and `wc_get_order_statuses()` do match in the end. WooCommerce sometimes uses the prefix, sometimes it doesn't. Internally, it needs sometimes to do things like this: https://github.com/woocommerce/woocommerce/blob/1e8bbb2800c2c446eb663e2c802acc39a27e5322/plugins/woocommerce/includes/wc-order-functions.php#L147-L152

`wc_get_order_statuses()` returns with the prefix. Looking into the data base the prefix is stored with the order. So I assume `get_status()` returns the status with the prefix. But did you had a closer look into it?

Reply:
@websupporter Very good point.

So:
- `wc_get_order_statuses()` includes the `wc-` prefix
- `$order->get_status()` doesn't (even `$order->set_status()` followed by `$order->get_status()` already strips it)

We do one of:
1. Append `wc-` when fetching status (`'wc-' . $payload->getOrder()->get_status()`).
2. Strip `wc-` from the list of all statuses, when the prefix exists.

These are similar, but not fully equivalent. If WC saves the status with `wc-`, then maybe I'd lean to doing the same. On the other hand, do all WC statuses, including some custom ones, have the `wc-` prefix? If there's a risk of "no, they don't", then 2. is safer, I guess.

What do you think?

Reply:
As discussed on Slack, option 2 seems safer: https://github.com/mailpoet/mailpoet/pull/4984/commits/6666e9e10c26911c35cb7abe9162099eeb11ac2b